### PR TITLE
use latest stable version of kube-network-policies for CI

### PR DESCRIPTION
**Pull request by aojea**
/kind failing-test
/kind flake
```release-note
NONE
```

Fixes: #https://github.com/kubernetes-sigs/kube-network-policies/issues/12

### CODE REVIEW

**Comment by aojea** [comment]
This is working now 
/assign @danwinship @BenTheElder

File: cluster/addons/kube-network-policies/kube-network-policies.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
       serviceAccountName: kube-network-policies
       containers:
       - name: kube-network-policies
-        image: registry.k8s.io/networking/kube-network-policies:v0.1.0
+        image: gcr.io/k8s-staging-networking/kube-network-policies:v20240423-88bdf40
```
Comment (BenTheElder) [inline cluster/addons/kube-network-policies/kube-network-policies.yaml:30]: can we publish an image?

we generally want to be pulling through registry.k8s.io

Comment (BenTheElder) [reply cluster/addons/kube-network-policies/kube-network-policies.yaml:30]: (and we generally want k/k on release tags)

Comment (aojea) [reply cluster/addons/kube-network-policies/kube-network-policies.yaml:30]: I wanted to get the CI green before cut another release, once the CI is stable I'll switch again to registry

**Pull request merged**